src/officehours_api/backends/zoom.py

```diff
@@ -0,0 +1,225 @@
+    refresh_token: str
+    expires_in: int
+    scope: str
+
+
+class Backend:
+    friendly_name = "Zoom"
+    base_url = 'https://zoom.us'
+    client_id = settings.ZOOM_CLIENT_ID
+    client_secret = settings.ZOOM_CLIENT_SECRET
+
+    @classmethod
+    def _get_client_auth_headers(cls) -> dict:
+        client = b64encode(bytes(f"{cls.client_id}:{cls.client_secret}", 'ascii')).decode('ascii')
+        return {
+            'Authorization': f"Basic {client}",
+            'Accept': 'application/json',
+        }
+
+    @classmethod
+    def _spend_authorization_code(cls, code: str) -> ZoomAccessToken:
+        resp = requests.post(
+            f'{cls.base_url}/oauth/token',
+            params={
+                'grant_type': 'authorization_code',
+                'code': code,
+                'redirect_uri': 'http://localhost:8003/callback/zoom/',
+            },
+            headers=cls._get_client_auth_headers(),
+        )
+        print(resp.request.url)
+        print(resp.json())
+        resp.raise_for_status()
+        # print(repr(resp.request.url))
+        # print(repr(resp.request.headers))
+        # print(resp)
+        # print(resp.text)
+        return resp.json()
+
+    @classmethod
+    def _get_access_token(cls, user: User) -> str:
+        zoom_meta = user.profile.backend_metadata['zoom']
+        if time() > zoom_meta['access_token_expires']:
+            resp = requests.post(
+                f'{cls.base_url}/oauth/token',
+                params={
+                    'grant_type': 'refresh_token',
+                    'refresh_token': zoom_meta['refresh_token'],
+                },
+                headers=cls._get_client_auth_headers(),
+            )
+            resp.raise_for_status()
+            token = resp.json()
+            zoom_meta.update({
+                'refresh_token': token['refresh_token'],
+                'access_token': token['access_token'],
+                'access_token_expires': time() - token['expires_in'] - 60,
+            })
+            user.profile.save()
+        return zoom_meta['access_token']
+
+    @classmethod
+    def _get_session(cls, user: User) -> requests.Session:
+        session = requests.Session()
+        session.headers.update({
+            'Authorization': f'Bearer {cls._get_access_token(user)}',
+            'Accept': 'application/json',
+            'Content-Type': 'application/json',
+        })
+        return session
+
+    @classmethod
+    def _create_meeting(cls, user: User) -> ZoomMeeting:
+        session = cls._get_session(user)
+        user_id = user.profile.backend_metadata['zoom']['user_id']
+        resp = session.post(
+            f'{cls.base_url}/v2/users/{user_id}/meetings',
+            json={
+                'topic': 'Remote Office Hours Queue Meeting',
+                'start_time': datetime.now().strftime('%Y-%m-%dT%H:%M:%SZ'),
+                'timezone': 'America/Detroit',
+                'agenda': 'Meeting agenda goes here',
```

Comment:
I removed it for now - we could add it back in later, but we would probably want to keep the agenda synchronized with ROHQ, and it could change at any time.